### PR TITLE
Add feedback files.

### PR DESCRIPTION
**Pull request by alexvail**
- Added Feedback Files: complete restful support
- Display feedback files in web interface

### CODE REVIEW

File: app/views/results/common/_test_selector.html.erb [hunk 1]
```diff
@@ -20,10 +20,10 @@
     </script>
 
   <select id="select_test_result_id" onchange="if ($(this).getValue() != '') { load_test_result($(this).getValue()); }">
```
Comment (alexvail) [inline app/views/results/common/_test_selector.html.erb]: This file can probably be deleted.

File: db/migrate/20160327163400_create_feedback_files.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CreateFeedbackFiles < ActiveRecord::Migration
+  def change
+    create_table :feedback_files do |t|
+      t.string :filename
+      t.binary :file_content
```
Comment (david-yz-liu) [inline db/migrate/20160327163400_create_feedback_files.rb]: This is good, but I think we'll need to store the content type as well now. https://gist.github.com/macek/610596

File: app/views/results/student/_student_panes.html.erb [hunk 3]
```diff
@@ -5,8 +5,8 @@
         <ul>
           <li><a href='#code_holder'><%= t('marker.source_code') %></a></li>
           <li><a href='#annotations_summary'><%= t('marker.annot_summary') %></a></li>
-          <% if can_show_test_results_tab?(assignment, submission) %>
-            <li><a href='#test_result_tab'><%= t('marker.test_results') %></a></li>
+          <% if can_show_feedback_files_tab?(submission) %>
+              <li><a href='#feedback_file_tab'><%= t('marker.feedback_files') %></a></li>
```
Comment (david-yz-liu) [inline app/views/results/student/_student_panes.html.erb]: Let's stay with the previous indentation.

File: app/views/results/common/_feedback_file_selector.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      <option value=""><%= I18n.t("test_result.no_files_available")%></option>
+    <% else %>
+      <% feedback_files.where(submission_id: submission_id).order("created_at DESC").each do |file| %>
+        <option value="<%= file.id %>">
+          <%= file.filename %>
+        </option>
+      <% end %>
+    <% end %>
+  </select>
+  <input type="button" onclick="if (jQuery('#select_feedback_file_id').val() != '') { load_feedback_file(jQuery('#select_feedback_file_id').val()); }"
```
Comment (david-yz-liu) [inline app/views/results/common/_feedback_file_selector.html.erb]: I find it interesting that both the `select` and this button have a `load_feedback_file` call. I wonder if we can simplify this so only the `select`'s onchange is used, and this button is removed?

File: app/views/results/edit.html.erb [hunk 5]
```diff
@@ -67,10 +67,9 @@
 <%= render partial: 'results/common/old_marks_view_functions',
            formats: [:html] %>
 
-<% # Javascript for test result window %>
-<%= render partial: 'assignments/test_result_window',
-           formats: [:html],
-           locals: { aid: @assignment.id } %>
+<% # Javascript for feedback file window %>
+<%= render partial: 'assignments/feedback_file_window',
```
Comment (david-yz-liu) [inline app/views/results/edit.html.erb]: Add "js" to the extension of the file, and make the format `[:js]`

File: app/views/results/common/_test_selector.html.erb (hunk 1 above)
Comment (david-yz-liu) [inline app/views/results/common/_test_selector.html.erb]: I agree, please do.

File: app/views/assignments/_feedback_file_window.erb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+<script>
+  function load_feedback_file(feedback_file_id) {
+    jQuery('select_feedback_file_id').prop( "disabled", true);
+    jQuery.ajax({
+      url: '<%= render_feedback_file_assignment_path(id: @assignment.id) %>',
+      type: 'GET',
+      data: 'feedback_file_id=' + feedback_file_id,
```
Comment (david-yz-liu) [inline app/views/assignments/_feedback_file_window.erb]: Use an object literal here `{feedback_file_id: feedback_file_id}`

File: app/views/assignments/_feedback_file_window.erb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      beforeSend: function(){
+        document.getElementById('working').style.display = '';
+      },
+      complete: function(){
+        document.getElementById('working').style.display = 'none';
+      },
+      success: function (data) {
+        feedback_file_html = data;
```
Comment (david-yz-liu) [inline app/views/assignments/_feedback_file_window.erb]: I don't think you need to use a temporary variable here

File: app/views/assignments/_feedback_file_window.erb [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<script>
+  function load_feedback_file(feedback_file_id) {
+    jQuery('select_feedback_file_id').prop( "disabled", true);
+    jQuery.ajax({
+      url: '<%= render_feedback_file_assignment_path(id: @assignment.id) %>',
+      type: 'GET',
+      data: 'feedback_file_id=' + feedback_file_id,
+      beforeSend: function(){
```
Comment (david-yz-liu) [inline app/views/assignments/_feedback_file_window.erb]: Put a space before the `{`

**Comment by david-yz-liu** [comment]
Thanks, @alexvail!

**Pull request merged**